### PR TITLE
Add python-nextgen client generator

### PR DESCRIPTION
**Pull request by feluelle**
- update open api generator cli version to 6.6.0
- update python clients version to 2.6.0

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: clients/gen/common.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-OPENAPI_GENERATOR_CLI_VER=5.4.0
+OPENAPI_GENERATOR_CLI_VER=6.6.0
```
Comment (feluelle) [inline clients/gen/common.sh:20]: We might want to do that in a separate PR which this one would depend on.

File: clients/gen/python-nextgen.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+find "${OUTPUT_DIR}/test" -type f -name \*.py -exec sed "${INPLACE_ARG[@]}" -e 's/client/airflow_client.client/g' {} +
+find "${OUTPUT_DIR}" -type f -a -name \*.md -exec sed "${INPLACE_ARG[@]}" -e 's/# client/# Apache Airflow Python Client/g' {} +
+find "${OUTPUT_DIR}" -type f -a -name \*.md -exec sed "${INPLACE_ARG[@]}" -e 's/import client/import airflow_client.client/g' {} +
+find "${OUTPUT_DIR}" -type f -a -name \*.md -exec sed "${INPLACE_ARG[@]}" -e 's/from client/from airflow_client.client/g' {} +
+find "${OUTPUT_DIR}" -type f -a -name \*.md -exec sed "${INPLACE_ARG[@]}" -e 's/getattr(client\.models/getattr(airflow_client.client.models/g' {} +
+
+# fix imports
+find "${OUTPUT_DIR}/client/" -type f -name \*.py -exec sed "${INPLACE_ARG[@]}" -e 's/import client\./import airflow_client.client./g' {} +
+find "${OUTPUT_DIR}/client/" -type f -name \*.py -exec sed "${INPLACE_ARG[@]}" -e 's/from client/from airflow_client.client/g' {} +
+find "${OUTPUT_DIR}/client/" -type f -name \*.py -exec sed "${INPLACE_ARG[@]}" -e 's/getattr(client\.models/getattr(airflow_client.client.models/g' {} +
```
Comment (feluelle) [inline clients/gen/python-nextgen.sh:62]: Do we really need that?

File: clients/gen/python.sh [hunk 3]
```diff
@@ -26,7 +26,7 @@ readonly CLEANUP_DIRS
 # shellcheck source=./clients/gen/common.sh
 source "${CLIENTS_GEN_DIR}/common.sh"
 
-VERSION=2.5.1
+VERSION=2.6.0
```
Comment (feluelle) [inline clients/gen/python.sh:29]: Does this need its own PR? Also, does that mean we need to regenerate the python client? That should have been set when we were releasing the 2.6.0 client.

Comment (pierrejeambrun) [reply clients/gen/python.sh:29]: I’m not at home, I will do a review next week, but I believe that what is missing from the client [release process](https://github.com/apache/airflow-client-python/blob/main/dev/README_RELEASE_CLIENT.md) is “Update python.sh in airflow/main with the new API version”. Changing the version for generating the client is part of the process, (see above) we can inspect the generated client, the API version is specified here for instance https://github.com/apache/airflow-client-python/blob/main/airflow_client/README.md#unknown which is correct.

PR to update the client release process is welcome. I think we have the same issue with the go client.

File: clients/gen/common.sh (hunk 1 above)
Comment (uranusjr) [reply clients/gen/common.sh:20]: What’s the benefit of doing that?

Comment (feluelle) [reply clients/gen/common.sh:20]: In a separate PR, you mean? The generator version when upgrading to a new major version contains breaking changes and since the version is shared between all our clients (python and go, too), I think it is worth its own PR.

File: clients/gen/python.sh (hunk 3 above)
Comment (feluelle) [reply clients/gen/python.sh:29]: Yeah, the version seems to be correct. I think I will move the change to a separate PR and document the missing piece in the release process docs.

File: clients/gen/python-nextgen.sh [hunk 4]
```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
```
Comment (pierrejeambrun) [inline clients/gen/python-nextgen.sh:1]: Why are we creating a new script and not updating `python.sh` ? Would we need to generate client with the old generator ? If so, should we update the release process for clients to use the appropriate script ?

File: clients/gen/python-nextgen.sh (hunk 2 above)
Comment (pierrejeambrun) [reply clients/gen/python-nextgen.sh:62]: I think we do because the file structure is a little bit different from the 'default' behavior. Can you try generating the client without it and running a few requests, to confirm if this is necessary ?

**Review by pierrejeambrun — COMMENTED** [review]
There are some breaking change in between `5.4.0` and `6.6.0` [release note](https://github.com/OpenAPITools/openapi-generator/releases), do you know how this would impact our generated code, and our end users ? Current generator is moved to `generator_prior`, are we safe to upgrade to nextgen?

Did you manage to run successful query with client generated with this new version, [test](https://github.com/apache/airflow-client-python/blob/main/dev/test_python_client.py) script is failing for me:
- It looks like we need to adjust the client 'dependencies', `pydantic` and `aenum` are now required by the generated client.
- `model` seems to have moved to `models`
- Certainly more breaking things that I didn't catch in my small experiment

Our user will be impacted, and some more adjustment needs to be made on the [python-client](https://github.com/apache/airflow-client-python) repository.

This also upgrades the generator version for `go` client. I think tests need to be made there as well to assess the impact.

**Comment by feluelle** [comment]
We can close this PR - `python-nextgen` has become `python`. I have created a new branch where I am testing `7.0.0-beta` and creating a PR as soon as I had time to test it.
